sky/backends/cloud_vm_ray_backend.py

```diff
@@ -1328,10 +1340,13 @@ def _provision(self,
             backoff = backend_utils.Backoff(_RETRY_UNTIL_UP_INIT_GAP_SECONDS)
             attempt_cnt = 1
             while True:
-                # RetryingVmProvisioner will retry within a cloud's regions
-                # first (if a region is not explicitly requested), then
-                # optionally retry on all other clouds (if
-                # backend.register_info() has been called).
+                # For on-demand instances, RetryingVmProvisioner will retry
+                # within the given region first, then optionally retry on all
+                # other clouds and regions (if backend.register_info()
+                # has been called).
+                # For spot instances, each provisioning request is made for a
+                # single zone and the provisioner will retry on all other
+                # clouds, regions, and zones.
```

Comment:
It's because of the AWS spot instance pricing. As AWS VMs in different zones may have different prices (even if they are in the same region), we need to make per-zone requests.

For GCP, SkyPilot already issues provisioning requests at the granularity of zones, but GCP applies the same pricing rule to spot VMs in the same region. At the time I implemented this PR, I didn't know that AWS is the only cloud that has such a weird pricing.

Reply:
This comment is very clear, thanks. Can we add it to the code as well?

Reply:
Added more detailed comments. PTAL.

Reply:
Maybe add something short here too to help readers of this module. E.g., "See optimizer.py#_make_launchables_for_valid_region_zones() for detailed reasons."

Reply:
Added.